File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int t : tangerine) {
            int tmp = map.getOrDefault(t, 0);
            map.put(t, tmp+1);
        }
        List<Integer> valueList = new ArrayList<>(map.values());
        valueList.sort(Collections.reverseOrder());
        for (int t : valueList) {
            k -= t;
            answer++;
            if (k <= 0) {
                break;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.py
def solution(k, tangerine):
    answer = 0
    info = dict()
    for t in tangerine:
        if t in info:
            info[t] += 1
        else:
            info[t] = 1
            
    tanger = list(info.items())
    tanger.sort(key=lambda x : -x[1])
    
    for t in tanger:
        k -= t[1]
        answer += 1
        if k <= 0:
            break
    return answer